### PR TITLE
Issue #40: Adding JUnit Tests to rcp-scion project

### PR DESCRIPTION
**Pull request by iachelini**
This commit adds the general ability to create JUnit tests and let them get executed during build time. This required:

- adding handpicked JUnit5 bundles from the Eclipse 2022/06 release
- adding jdt.junit runtime bundle
- adding tychos surefire plugin to the parent pom
- Fixing JUnit5 (5.8.1), hamcrest (2.2), objenesis (3.2) and mockito (4.0.0) to fit to each other
- Adding a sample test mocking a final class (GSon) and using hamcrest matchers
- Fixing classpath entries to match jdk11

REVIEW: ozohje, ruadrian

### CODE REVIEW

File: ch.sbb.scion.rcp.feature.test/feature.xml [hunk 1]
```diff
@@ -0,0 +1,165 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<feature
+      id="ch.sbb.scion.rcp.feature.test"
+      label="Test"
+      version="1.0.0.qualifier"
+      provider-name="SBB">
+
+   <description url="http://www.example.com/description">
```
Comment (ozohje) [inline ch.sbb.scion.rcp.feature.test/feature.xml:8]: Are these 3 fields(description, copyright and license) mandatory for a feature.xml? Or just added as placeholders for future completion?

File: ch.sbb.scion.rcp.microfrontend.test/src/ch/sbb/scion/rcp/microfrontend/internal/gson/SetObjectTypeAdapterFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  void create_shouldReturnNull_forTypeTokenNotSet() {
+    TypeToken<Long> longToken = TypeToken.get(Long.class);
+    TypeAdapter<Long> typeAdapter = factory.create(gson, longToken);
+
+    assertNull(typeAdapter);
+  }
+
+  @Test
+  void create_shouldReturnTypeAdapter_forTypeTokenAsHashSet() {
+    TypeToken<HashSet<String>> hasSetToken = (TypeToken<HashSet<String>>) TypeToken.getParameterized(HashSet.class, String.class);
```
Comment (ozohje) [inline ch.sbb.scion.rcp.microfrontend.test/src/ch/sbb/scion/rcp/microfrontend/internal/gson/SetObjectTypeAdapterFactoryTest.java]: Little typo in the object instance name, probably you meant "hashSetToken" here.

**Review by ozohje — COMMENTED** [review]
Very nice work with setting up our plugins to be JUnit-testable, would take probably at least a weak for me to get it right... my review is valid mostly for the added tests where I managed to find one little typo and I was kind of shy to point even that one out.
Regarding the pom.xml changes I highly trust that you added the correct settings, unfortunately I'm not really a good candidate to give a valid review for those, for me that's more like a very nice reference in case I need to do something similar in the future.

File: ch.sbb.scion.rcp.feature.test/feature.xml (hunk 1 above)
Comment (iachelini) [reply ch.sbb.scion.rcp.feature.test/feature.xml:8]: You are right. I should either provide useful information or omit it entirely.

File: pom.xml [hunk 3]
```diff
@@ -9,20 +9,53 @@
 	<packaging>pom</packaging>
 
 	<properties>
+		<java.version>11</java.version>
+		<maven.compiler.source>${java.version}</maven.compiler.source>
```
Comment (mentamas) [inline pom.xml:13]: Doesn't the maven.compiler.release contain/replace the compiler.source and compiler.target? https://www.baeldung.com/maven-java-version#java9

File: ch.sbb.scion.rcp.microfrontend.test/src/ch/sbb/scion/rcp/microfrontend/internal/gson/SetObjectTypeAdapterFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Project: RCS - Rail Control System
+ *
+ * © Copyright by SBB AG, Alle Rechte vorbehalten
+ */
+package ch.sbb.scion.rcp.microfrontend.internal.gson;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (mentamas) [inline ch.sbb.scion.rcp.microfrontend.test/src/ch/sbb/scion/rcp/microfrontend/internal/gson/SetObjectTypeAdapterFactoryTest.java:8]: Please fix the wildcard imports.

File: ch.sbb.scion.rcp.microfrontend.test/src/ch/sbb/scion/rcp/microfrontend/internal/gson/SetObjectTypeAdapterFactoryTestIT.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Project: RCS - Rail Control System
+ *
+ * © Copyright by SBB AG, Alle Rechte vorbehalten
+ */
+package ch.sbb.scion.rcp.microfrontend.internal.gson;
+
+import static org.hamcrest.MatcherAssert.*;
```
Comment (mentamas) [inline ch.sbb.scion.rcp.microfrontend.test/src/ch/sbb/scion/rcp/microfrontend/internal/gson/SetObjectTypeAdapterFactoryTestIT.java:8]: Please fix the wildcard imports.

File: ch.sbb.scion.rcp.workbench.test/src/ch/sbb/scion/rcp/workbench/view/MicrofrontendViewEditorInputTest.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Project: RCS - Rail Control System
+ *
+ * © Copyright by SBB AG, Alle Rechte vorbehalten
+ */
+package ch.sbb.scion.rcp.workbench.view;
+
+import static org.hamcrest.MatcherAssert.*;
```
Comment (mentamas) [inline ch.sbb.scion.rcp.workbench.test/src/ch/sbb/scion/rcp/workbench/view/MicrofrontendViewEditorInputTest.java:8]: Please fix the wildcard imports.

File: ch.sbb.scion.rcp.workbench.test/META-INF/MANIFEST.MF [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+Fragment-Host: ch.sbb.scion.rcp.workbench
+Automatic-Module-Name: ch.sbb.scion.rcp.workbench.test
+Bundle-RequiredExecutionEnvironment: JavaSE-11
+Bundle-Vendor: SBB and Contributors
+Require-Bundle: org.junit.jupiter.api,
+ org.mockito.mockito-core,
+ org.mockito.junit-jupiter,
+ wrapped.org.mockito.mockito-inline,
+ org.hamcrest,
+ net.bytebuddy.byte-buddy,
```
Comment (mentamas) [inline ch.sbb.scion.rcp.workbench.test/META-INF/MANIFEST.MF:15]: Please define only the direct dependencies of the test fragment as Required Bundle and remove the rest.

File: ch.sbb.scion.rcp.microfrontend.test/META-INF/MANIFEST.MF [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+Bundle-Vendor: SBB and Contributors
+Fragment-Host: ch.sbb.scion.rcp.microfrontend
+Automatic-Module-Name: ch.sbb.scion.rcp.microfrontend.test
+Bundle-RequiredExecutionEnvironment: JavaSE-11
+Require-Bundle: org.junit.jupiter.api,
+ org.mockito.mockito-core,
+ org.mockito.junit-jupiter,
+ wrapped.org.mockito.mockito-inline,
+ org.hamcrest,
+ net.bytebuddy.byte-buddy,
```
Comment (mentamas) [inline ch.sbb.scion.rcp.microfrontend.test/META-INF/MANIFEST.MF:15]: Please define only the direct dependencies of the test fragment as Required Bundle and remove the rest.

**Review by mentamas — COMMENTED** [review]
build.out → I guess it was pushed by accident. Please remove it.

File: pom.xml (hunk 3 above)
Comment (iachelini) [reply pom.xml:13]: removed target and source then

**Review by ruadrian — APPROVED** [review]

**Pull request merged**